app/scripts/background.js

```diff
@@ -724,22 +756,60 @@ export function setupController(
           }
         });
       }
-      connectExternal(remotePort);
+      connectExternalLegacy(remotePort);
     }
   };
 
   // communication with page or other extension
-  connectExternal = (remotePort) => {
+  connectExternalLegacy = (remotePort) => {
+    const portStream =
+      overrides?.getPortStream?.(remotePort) || new PortStream(remotePort);
+    controller.setupUntrustedCommunicationLegacy({
+      connectionStream: portStream,
+      sender: remotePort.sender,
+    });
+  };
+
+  connectExternalCaip = async (remotePort) => {
+    if (metamaskBlockedPorts.includes(remotePort.name)) {
+      return;
+    }
+
+    // this is triggered when a new tab is opened, or origin(url) is changed
+    if (remotePort.sender && remotePort.sender.tab && remotePort.sender.url) {
+      const tabId = remotePort.sender.tab.id;
+      const url = new URL(remotePort.sender.url);
+      const { origin } = url;
+
+      trackDappView(
+        remotePort,
+        controller.preferencesController,
+        controller.permissionController,
+      );
+
+      // TODO: remove this when we separate the legacy and multichain rpc pipelines
```

Comment:
Curious to know what this refers to, i.e. what does it mean to "separate the legacy and multichain RPC pipelines", and how is this `requestAccountTabIds` state related?

Reply:
@Gudahtt we are intending to create a separate JSON RPC pipeline for the new API ([ticket here](https://github.com/MetaMask/MetaMask-planning/issues/2527)) for long term maintainability

Reply:
@Gudahtt just want to confirm you've seen and understood this roadmap?